spot-client/src/spot-remote/ui/components/dial-pad/DialPad.test.js

```diff
@@ -16,47 +16,103 @@ describe('DialPad', () => {
+
+        expect(dialPad.find('input').instance().value).toBe('(222) 333-4444');
+    });
+
+    test('formats international phone numbers', () => {
+        // Long press on 0 to produce +
+        jest.useFakeTimers();
+        dialPad.find('#dial-button-0').simulate('mousedown');
+        jest.runAllTimers();
+
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+
+        dialPad.find(`#dial-button-${5}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${5}`).simulate('touchstart');
+
+        dialPad.find(`#dial-button-${6}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${6}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${6}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${6}`).simulate('touchstart');
+
+        dialPad.find(`#dial-button-${7}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${7}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${7}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${7}`).simulate('touchstart');
+
+        expect(dialPad.find('input').instance().value).toBe('+44 55 6666 7777');
     });
 
     test('deleting numbers', () => {
-        for (let i = 1; i < 4; i++) {
-            dialPad.find(`#dial-button-${i}`).simulate('mousedown');
-        }
+        dialPad.find(`#dial-button-${2}`).simulate('mousedown');
+        dialPad.find(`#dial-button-${2}`).simulate('mousedown');
+        dialPad.find(`#dial-button-${2}`).simulate('mousedown');
 
         dialPad.find('.backspace').simulate('click');
-        expect(dialPad.find('input').instance().value).toBe('12');
+        expect(dialPad.find('input').instance().value).toBe('22');
 
         dialPad.find('.backspace').simulate('click');
-        expect(dialPad.find('input').instance().value).toBe('1');
+        expect(dialPad.find('input').instance().value).toBe('2');
 
         dialPad.find('.backspace').simulate('click');
         expect(dialPad.find('input').instance().value).toBe('');
     });
 
-    test('submitting the entered number', () => {
-        dialPad.find('#dial-button-1').simulate('mousedown');
+    test('submitting a valid phone number in the E.164 format', () => {
+        dialPad.find(`#dial-button-${2}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${3}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+
+        dialPad.find(`#dial-button-${3}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${3}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${3}`).simulate('touchstart');
+
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
+        dialPad.find(`#dial-button-${4}`).simulate('touchstart');
```

Comment:
Maybe instead have a function to loop over a string?

Reply:
yeah